watchdog/watchdog_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package watchdog
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/keybase/go-logging"
+	"github.com/keybase/go-updater/process"
+	"github.com/keybase/go-updater/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var testLog = &logging.Logger{Module: "test"}
+
+func TestWatch(t *testing.T) {
+	procProgram1 := procProgram(t, "testWatch1", 60)
+	procProgram2 := procProgram(t, "testWatch2", 60)
+
+	delay := 10 * time.Millisecond
+
+	err := Watch([]Program{procProgram1, procProgram2}, delay, testLog)
+	require.NoError(t, err)
+
+	matcher1 := process.NewMatcher(procProgram1.Path, process.PathEqual, testLog)
+	procs1, err := process.FindProcesses(matcher1, time.Second, 200*time.Millisecond, testLog)
+	require.NoError(t, err)
+	assert.Equal(t, 1, len(procs1))
+
+	matcher2 := process.NewMatcher(procProgram2.Path, process.PathEqual, testLog)
+	procs2, err := process.FindProcesses(matcher2, time.Second, 200*time.Millisecond, testLog)
+	require.NoError(t, err)
+	assert.Equal(t, 1, len(procs2))
+	proc2 := procs2[0]
+
+	err = process.TerminatePID(proc2.Pid(), time.Millisecond, testLog)
+	require.NoError(t, err)
+
+	time.Sleep(2 * delay)
+
+	// Check for restart
+	procs2After, err := process.FindProcesses(matcher2, time.Second, time.Millisecond, testLog)
+	require.NoError(t, err)
+	assert.Equal(t, 1, len(procs2After))
+}
+
+func cleanupProc(cmd *exec.Cmd, procPath string) {
+	if cmd != nil && cmd.Process != nil {
+		_ = cmd.Process.Kill()
+	}
+	_ = os.Remove(procPath)
+}
+
+// procProgram returns a testable unique program at a temporary location
+func procProgram(t *testing.T, name string, delaySeconds int) Program {
+	// Copy sleep executable to tmp
+	procPath := filepath.Join(os.TempDir(), name)
+	err := util.CopyFile("/bin/sleep", procPath, testLog)
```

Comment:
actually we have a go test binary for stuff like this.. changed to use this one instead